torch/_inductor/ir.py

```diff
@@ -2749,7 +2749,9 @@ def indexer(index):
             assert len(index) == len(self.stride) == len(self.size)
             result = self.offset
             for idx, stride, sz in zip(index, self.stride, self.size):
-                if sz != 1:
+                # If there is indirect indexing we have to perform the
+                # idx wrapping and check that 0 <= idx < 1 checks
+                if not (sz == 1 and free_symbol_is_type(idx, SymT.TMP)):
```

Comment:
See this change. I think it really is necessary for correctness. It should almost never be hit though (just when we have an indirect indexing that we were not able to constant propagate into a dimension of size 1).